### PR TITLE
Update CHANGES.txt with 3.11.0-RC1 release notes

### PR DESCRIPTION
**Pull request by rafi-kamal**

### CODE REVIEW

**Comment by rafi-kamal** [comment]
Also @ObsidianMinor to review :)

File: CHANGES.txt [hunk 1]
```diff
@@ -1,3 +1,61 @@
...
+  * Publish ProGuard config for javalite
+  * Fix for StrictMode disk read violation in ExtensionRegistryLite
+  * Include part of the ByteString's content in its toString().
+  * Include unknown fields when merging proto3 messages in Java lite builders
+
+  Python
+  * Clarify the behavior of text_format.Parse/Merge on non-repeated fields when the input proto is non-empty
+  * Add float_precision option in json format printer
+  * Improve error message when using unbuffered streams with delimited messages
+  * Optionally print bytes fields as messages, if possible
```
Comment (anandolee) [inline CHANGES.txt]: Optionally print bytes fields as messages in unknown fields, if possible

File: CHANGES.txt [hunk 2]
```diff
@@ -1,3 +1,61 @@
...
+
+  Java
+  * Remove the usage of MethodHandle, so that Android users prior to API version 26 can use protobuf-java
+  * Improved Javadoc for ByteString
+  * Publish ProGuard config for javalite
+  * Fix for StrictMode disk read violation in ExtensionRegistryLite
+  * Include part of the ByteString's content in its toString().
+  * Include unknown fields when merging proto3 messages in Java lite builders
+
+  Python
```
Comment (anandolee) [inline CHANGES.txt]: 1, Python ads code yellow  change should be included
2, Remove for following ( too minor that I think no need to be listed in changes):
          * Clarify the behavior of text_format.Parse/Merge on non-repeated fields when the input proto is non-empty
          * Improve error message when using unbuffered streams with delimited messages

File: CHANGES.txt [hunk 3]
```diff
@@ -1,3 +1,61 @@
...
+  PHP
+  * Avoid too much overhead in layout_init (#6716)
+  * Lazily Create Singular Wrapper Message (#6833)
+
+  Ruby
+  * Ruby lazy wrappers optimization (#6797)
+
+  C#
+  * (RepeatedField): Capacity property to resize the internal array (#6530)
+  * C# Proto2 feature : Finale (#5936)
```
Comment (ObsidianMinor) [inline CHANGES.txt]: Should probably point to all PRs
```suggestion
  * Experimental proto2 support is now officially available (#4642, #5183, #5350, #5936)
    * Getting started doc: https://github.com/protocolbuffers/protobuf/blob/master/docs/csharp/proto2.md
```

File: CHANGES.txt [hunk 4]
```diff
@@ -1,3 +1,61 @@
...
+  * Add length checks to ExtensionCollection (#6759)
+  * Optimize parsing of some primitive and wrapper types (#6843)
+  * Document proto2 feature implementations in Google.Protobuf (#6499)
```
Comment (ObsidianMinor) [inline CHANGES.txt]: Could be removed if you add the "getting started doc" line.

File: CHANGES.txt (hunk 2 above)
Comment (rafi-kamal) [reply CHANGES.txt]: Done.

File: CHANGES.txt (hunk 1 above)
Comment (rafi-kamal) [reply CHANGES.txt]: Done.

File: CHANGES.txt (hunk 4 above)
Comment (rafi-kamal) [reply CHANGES.txt]: Done, thanks!

File: CHANGES.txt [hunk 5]
```diff
@@ -1,3 +1,60 @@
...
+  * (RepeatedField): Capacity property to resize the internal array (#6530)
+  * Experimental proto2 support is now officially available (#4642, #5183, #5350, #5936)
+  * Getting started doc: https://github.com/protocolbuffers/protobuf/blob/master/docs/csharp/proto2.md
+  * Add length checks to ExtensionCollection (#6759)
+  * Optimize parsing of some primitive and wrapper types (#6843)
+  * Use 3 parameter Encoding.GetString for default string values (#6828)
+  * Change _Extensions property to normal body rather than expression (#6856)
+
+  Objective C
+  * Remove OSReadLittle* due to alignment requirements (#6678)
```
Comment (thomasvl) [inline CHANGES.txt]: Might make more sense to folks to say something like: "Fixed unaligned reads for 32bit arm with newer Xcode versions (#6678)"

Comment (rafi-kamal) [reply CHANGES.txt]: Done.

**Pull request merged**